src/src_main/crypto_tc.c

```diff
@@ -53,6 +53,7 @@ int32_t Crypto_TC_ApplySecurity(const uint8_t* p_in_frame, const uint16_t in_fra
     uint32_t encryption_cipher;
     uint8_t ecs_is_aead_algorithm;
     int i;
+    uint32_t padding_needed = 0;
```

Comment:
It could be larger.  There's a possibility that other cipers might have blocks larger than 16.
So if we have blocks of 32, we would need 2 bytes to express the amount of padding we could possibly need.
The standard says the max length the sh_plf could be is 3 bytes.  So that's why I used the uint32_t.

As I'm typing this, I'm second guessing myself.  So I'll let others weigh in on this one as well.  But I 'think' it is ok.

Reply:
According to the way padding_needed was set below, I don't think it can be larger with this current implementation.

Reply:
Ignore this --- apparently we're hard-coding CryptoLib to support up to 3 bytes for the pad length field.